metal/v1/model_device.go

```diff
@@ -1,1680 +1,71 @@
-	SpotPriceMax *float32 `json:"spot_price_max,omitempty"`
-	SshKeys      []Href   `json:"ssh_keys,omitempty"`
+	SpotPriceMax float32 `json:"spot_price_max,omitempty"`
+	SshKeys      []Href  `json:"ssh_keys,omitempty"`
 	// The current state the instance is in.  * When an instance is initially created it will be in the `queued` state until it is picked up by the provisioner. * Once provisioning has begun on the instance it's state will move to `provisioning`. * When an instance is deleted, it will move to `deprovisioning` state until the deprovision is completed and the instance state moves to `deleted`. * If an instance fails to provision or deprovision it will move to `failed` state. * Once an instance has completed provisioning it will move to `active` state. * If an instance is currently powering off or powering on it will move to `powering_off` or `powering_on` states respectively.  * When the instance is powered off completely it will move to the `inactive` state. * When an instance is powered on completely it will move to the `active` state. * Using the reinstall action to install a new OS on the instance will cause the instance state to change to `reinstalling`. * When the reinstall action is complete the instance will move to `active` state.
-	State   *string  `json:"state,omitempty"`
+	State   string   `json:"state,omitempty"`
 	Storage *Storage `json:"storage,omitempty"`
 	// Switch short id. This can be used to determine if two devices are connected to the same switch, for example.
-	SwitchUuid *string  `json:"switch_uuid,omitempty"`
+	SwitchUuid string   `json:"switch_uuid,omitempty"`
 	Tags       []string `json:"tags,omitempty"`
 	// When the device will be terminated. If you don't supply timezone info, the timestamp is assumed to be in UTC.  This is commonly set in advance for ephemeral spot market instances but this field may also be set with on-demand and reservation instances to automatically delete the resource at a given time. The termination time can also be used to release a hardware reservation instance at a given time, keeping the reservation open for other uses.  On a spot market device, the termination time will be set automatically when outbid.
-	TerminationTime *time.Time `json:"termination_time,omitempty"`
-	UpdatedAt       *time.Time `json:"updated_at,omitempty"`
-	User            *string    `json:"user,omitempty"`
-	Userdata        *string    `json:"userdata,omitempty"`
-	Volumes         []Href     `json:"volumes,omitempty"`
+	TerminationTime time.Time `json:"termination_time,omitempty"`
+	UpdatedAt       time.Time `json:"updated_at,omitempty"`
+	User            string    `json:"user,omitempty"`
+	Userdata        string    `json:"userdata,omitempty"`
+	Volumes         []Href    `json:"volumes,omitempty"`
 	// Hostname used to connect to the instance via the SOS (Serial over SSH) out-of-band console.
-	Sos                  *string `json:"sos,omitempty"`
-	AdditionalProperties map[string]interface{}
```

Comment:
`AdditionalProperties` disappeared, but that could be a configuration issue (with openapi-generator we had to tell the generator to generate spec-compliant code with AdditionalProperties support, and it's possible a similar config option is needed with swagger-codegen)